Fix/database settings spin box bug

jNullj opened the pull request:
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )

Found a bug while adding another feature at the Database Settings widget...
The database settings for Max items history and Max history size are SpinBox enabled when loaded from metadata as disabled.
Also for Max history size, the size is ridiculously high, it says MB but the value loaded from default is in Bytes.
This pull request fixes both minor UI issues.

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )
Before the bug when opening a database where the checkbox is unchecked:
- ❎ Default size in MB
- ❎ Enable state of SpinBox
![image](https://user-images.githubusercontent.com/15849761/218258605-6ec52f9f-7521-4113-8aec-0254bd0bff7b.png)
After the fix:
- ✅ Default size in MB
- ✅ Enable state of SpinBox
![image](https://user-images.githubusercontent.com/15849761/218258573-078fa600-ef15-43f9-90fe-e7695708b79e.png)



## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )

Tested on linux manually against latest release (archlinux).
Also run coverage with 100% pass


## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change that fixes an issue)
- ❎ New feature (change that adds functionality)
- ❎ Breaking change (causes existing functionality to change)
- ❎ Refactor (significant modification to existing code)
- ❎ Documentation (non-code change)

jNullj committed:
Fix DefaultHistoryMaxSize set in UI

The default value in the Metadata is saved in Bytes while the UI shows the size in MB.
Old code used the default in metadata as is, if database max size is unset the gui presents the default in bytes insted of MB which results in a huge number.
This commit fixes this by adding the calculation to transform the value to MB.

jNullj committed:
Fix DatabaseSettings UI SpinBoxs enable state

SpinBoxes for items with disabled checkbox should not be enabled.
If the user saves those settings as dissabled then reload the database the UI defaults to enable the SpinBox.
This commit fixes this by setting enable to false if the setting checkbox is not checked.

jNullj commented:
CI seems to fail at build job for ubuntu in one of the tests. but the test failed seems unrelated to the changes and all other builds and the test coverage seems to success...
Can you re-trigger the CI pipe?

dmaslenko commented on src/gui/dbsettings/DatabaseSettingsWidgetGeneral.cpp:
```diff
@@ -57,7 +57,7 @@ void DatabaseSettingsWidgetGeneral::initialize()
         m_ui->historyMaxSizeCheckBox->setChecked(true);
     } else {
-        m_ui->historyMaxSizeSpinBox->setValue(Metadata::DefaultHistoryMaxSize);
+        m_ui->historyMaxSizeSpinBox->setValue(qRound(Metadata::DefaultHistoryMaxSize / qreal(1048576)));
```
For better readability I would define as `1024 * 1024`. It will not impact on runtime performance because the compiler will multiply it on build time.

jNullj replied:
Didn't know compilers can do that, assumed it would be best to keep the result as i saw it already written that way on line 56.
I will fix this shortly

dmaslenko commented on src/gui/dbsettings/DatabaseSettingsWidgetGeneral.cpp:
```diff
@@ -50,6 +50,7 @@ void DatabaseSettingsWidgetGeneral::initialize()
         m_ui->historyMaxItemsCheckBox->setChecked(true);
     } else {
         m_ui->historyMaxItemsSpinBox->setValue(Metadata::DefaultHistoryMaxItems);
+        m_ui->historyMaxItemsSpinBox->setEnabled(false);
```
Is it easy to test it in GUI tests?

droidmonkey replied:
doesn't seem useful, this is a very minor UI setting. I also could not replicate the original problem reported of the max history items spin box being enabled even though the checkbox was unchecked.

jNullj replied:
> doesn't seem useful, this is a very minor UI setting. I also could not replicate the original problem reported of the max history items spin box being enabled even though the checkbox was unchecked.

To replicate you could make a new db, disable both settings then close keepassxc and relaunch into the settings. I can replicate 100% of the time on linux at current release.

I could still add the test altho i also think its a bit too much for such a minor bug. I could also see that this code is run in coverage already.
![image](https://user-images.githubusercontent.com/15849761/218336160-7a134d15-ea15-452b-b897-2b04d102f51f.png)

droidmonkey replied:
Ahhh you have to restart the program, missing detail :-)

jNullj committed:
Improve readability

Values in DatabaseSettingsWidgetGeneral to convert from bytes to MB are written a more reabable way.
